### PR TITLE
Move delegate handler and related code to i.b.admin for easier reuse"

### PR DESCRIPTION
**Pull request by adleong**
Some minor refactoring to make DelegateHandler and related code independent of linkerd.

### CODE REVIEW

File: admin/src/main/scala/io/buoyant/admin/StaticFilter.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "js" -> MediaType.Javascript,
+    "json" -> MediaType.Json,
+    "png" -> MediaType.Png,
+    "svg" -> "image/svg+xml",
+    "txt" -> MediaType.Txt
+  )
+
+  def apply(req: Request, svc: Service[Request, Response]) = {
+    svc(req).flatMap { res =>
+      val ext = req.path.split('.').lastOption.getOrElse("")
```
Comment (gtcampbell) [inline admin/src/main/scala/io/buoyant/admin/StaticFilter.scala]: I don't think this is correct - `"/foo/bar".split(".").lastOption` is `Some("/foo/bar")`. Can you switch this to use `req.fileExtension`?

File: linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/LinkerdAdmin.scala [hunk 2]
```diff
@@ -4,12 +4,24 @@ import com.twitter.app.App
...
 
 class LinkerdAdmin(app: App, linker: Linker, config: LinkerConfig) extends Admin(app) {
 
+  private[this] val dtabs: Future[Map[String, Dtab]] =
+    Future.value(
+      linker.routers.map { router =>
+      val RoutingFactory.BaseDtab(dtab) = router.params[RoutingFactory.BaseDtab]
+      router.label -> dtab()
+    }.toMap
+    )
```
Comment (gtcampbell) [inline linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/LinkerdAdmin.scala]: lines 20-22 should be indented one more level, I think

**Comment by gtcampbell** [comment]
LGTM other than my two cleanup comments

**Comment by olix0r** [comment]
:star:

**Pull request merged**